File: frontend/src/types/widgetSettings.ts
export interface WidgetConfig {
  key: string;
  visible: boolean;
  order: number;
}

export interface WidgetSettings {
  id: number;
  user_id: number;
  settings: string;
  created_at: string;
  updated_at: string;
}

export interface UpdateWidgetSettingsRequest {
  settings: WidgetConfig[];
}

File: backend/internal/model/widget_settings.go
package model

import "time"

// WidgetSettings はダッシュボードのウィジェット表示設定。
// ユーザーごとにウィジェットの表示/非表示と並び順をJSON形式で保存する。
type WidgetSettings struct {
	ID        uint      `json:"id"`
	UserID    uint      `json:"user_id"`
	Settings  string    `json:"settings"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
